File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
 
public class Solution {
    public List<List<int>> LevelOrder(TreeNode root) {
        List<List<int>> levelOrderList = new List<List<int>>();
        Queue<TreeNode> queue = new Queue<TreeNode>();

        if(root !=null)
            queue.Enqueue(root);

        while(queue.Count > 0){
            List<int> levelList = new List<int>();
            int len = queue.Count;
            for(int i=0; i< len ;i++){
                TreeNode node = queue.Dequeue();
                levelList.Add(node.val);
                if(node.left != null )
                    queue.Enqueue(node.left);
                if(node.right != null)
                    queue.Enqueue(node.right);
            }
            levelOrderList.Add(levelList);
        }
      return levelOrderList;
    }
}

File: Data Structures & Algorithms/invert-a-binary-tree/submission-0.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {TreeNode}
     */
    invertTree(root: TreeNode | null): TreeNode {

        if(root === null) return null;

        let tempNode: TreeNode | null = null;
        tempNode = root.left;
        root.left = root.right;
        root.right = tempNode;
        this.invertTree(root.left);
        this.invertTree(root.right);

        return root;
    }
}
